src/engine.py

```
@@ -2982,20 +3145,93 @@ async def _handle_outbound_answered(self, channel_id: str, channel: Dict[str, An
+            )
+            return
+
+        # Track for early-failure correlation (answer could race with mapping).
+        meta = dict(meta)
+        meta["channel_id"] = channel_id
+        self._outbound_attempt_meta_by_attempt_id[attempt_id] = meta
+        self._outbound_attempt_meta_by_channel_id[channel_id] = meta
+        try:
+            await self.outbound_store.set_attempt_channel(attempt_id, channel_id)
+        except Exception:
+            pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)